### PR TITLE
Feature/radius test

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Allow to set the username and password that pfstat will use to test the eduroam and radius source

# Impacts
pfstats

# Delete branch after merge
YES

# Checklist
- [n ] Document the feature
- [n ] Add unit tests
- [n ] Add acceptance tests (TestLink)

# NEWS file entries

## Enhancements
* Add a test user and password to test the Eduroam and RADIUS source

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm [hunk 1]
```diff
@@ -126,7 +126,25 @@ has_field 'monitor',
...
+   default => pf::Authentication::Source::RADIUSSource->meta->get_attribute('testUser')->default,
+);
 
+has_field 'testPassword',
+  (
+   type => 'ObfuscatedText',
+   label => 'Test Password',
+   element_class => ['input-small'],
+   tags => { after_element => \&help,
+             help => 'If you monitor, what is the password to test the RADIUS source' },
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm]: The password to use for testing when monitoring the RADIUS source

File: html/pfappserver/lib/pfappserver/Form/Config/Source/RADIUS.pm [hunk 2]
```diff
@@ -60,7 +60,26 @@ has_field 'monitor',
...
+   default => '',
+);
+
+has_field 'testPassword',
+  (
+   type => 'ObfuscatedText',
+   label => 'Test Password',
+   element_class => ['input-small'],
+   tags => { after_element => \&help,
+             help => 'If you monitor, what is the password to test the RADIUS source' },
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/RADIUS.pm]: The password to use for testing when monitoring the RADIUS source

File: html/pfappserver/lib/pfappserver/Form/Config/Source/RADIUS.pm [hunk 3]
```diff
@@ -60,7 +60,26 @@ has_field 'monitor',
...
-   default => pf::Authentication::Source::RADIUSSource->meta->get_attribute('monitor')->default,
+   default => '',
+);
+has_field 'testUser',
+  (
+   type => 'Text',
+   label => 'Test User',
+   element_class => ['input-small'],
+   tags => { after_element => \&help,
+             help => 'If you monitor, what is the username to test the RADIUS source' },
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/RADIUS.pm]: The username to use for testing when monitoring the RADIUS source

File: html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm [hunk 4]
```diff
@@ -126,7 +126,25 @@ has_field 'monitor',
              help => 'Do you want to monitor this source?' },
    default => pf::Authentication::Source::EduroamSource->meta->get_attribute('monitor')->default,
 );
+has_field 'testUser',
+  (
+   type => 'Text',
+   label => 'Test User',
+   element_class => ['input-small'],
+   tags => { after_element => \&help,
+             help => 'If you monitor, what is the username to test the RADIUS source' },
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm]: The username to use for testing when monitoring the RADIUS source

File: go/stats/main.go [hunk 5]
```diff
@@ -167,43 +171,46 @@ type eduroamtype struct{}
...
-	} else {
-		StatsdClient.Count("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"1", 1)
-		if response.Code == radius.CodeAccessAccept {
-			fmt.Println("Accepted")
+		if err != nil {
+			StatsdClient.Gauge("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"1", 0)
 		} else {
-			fmt.Println("Denied")
+			StatsdClient.Count("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"1", 1)
+			if response.Code == radius.CodeAccessAccept {
```
Comment (julsemaan) [inline go/stats/main.go:187]: Would be useful to record whether it was accepted or not for more in-depth troubleshooting if needed

File: go/stats/main.go [hunk 6]
```diff
@@ -167,43 +171,46 @@ type eduroamtype struct{}
...
+		packet = radius.New(radius.CodeAccessRequest, []byte(source.(pfconfigdriver.AuthenticationSourceEduroam).RadiusSecret))
+		UserName_SetString(packet, source.(pfconfigdriver.AuthenticationSourceEduroam).TestUser)
+		UserPassword_SetString(packet, source.(pfconfigdriver.AuthenticationSourceEduroam).TestPassword)
+		response, err = client.Exchange(ctx, packet, source.(pfconfigdriver.AuthenticationSourceEduroam).Server2Address+":1812")
+		if err != nil {
+			StatsdClient.Gauge("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"2", 0)
 		} else {
-			fmt.Println("Denied")
+			StatsdClient.Gauge("source."+source.(pfconfigdriver.AuthenticationSourceEduroam).Type+"."+source.(pfconfigdriver.AuthenticationSourceEduroam).PfconfigHashNS+"2", 1)
+			if response.Code == radius.CodeAccessAccept {
```
Comment (julsemaan) [inline go/stats/main.go:204]: Same as my comment above

**Comment by fdurand** [comment]
Needs to be reworked, instead of sending a Access Request we should only sent a Status Server.

File: html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm [hunk 7]
```diff
@@ -126,7 +126,25 @@ has_field 'monitor',
...
+  (
+   type => 'Text',
+   label => 'Test User',
+   element_class => ['input-small'],
+   tags => { after_element => \&help,
+             help => 'The username to use for testing when monitoring the RADIUS source' },
+   default => '',
+);
 
+has_field 'testPassword',
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm:139]: Add testPassword to the fields_to_mask list in the sources.
https://github.com/inverse-inc/packetfence/blob/devel/lib/pf/UnifiedApi/Controller/Config/Sources.pm#L246

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by fdurand** [comment]
This PR is useless now, we use status-check to test the radius sources.

**Pull request closed without merging**